apps/service_providers/llm_service/runnables.py

```diff
@@ -280,6 +280,24 @@ def invoke(self, input: str, config: RunnableConfig | None = None) -> ChainOutpu
         if thread_id:
             input_dict["thread_id"] = thread_id
 
+        # Set attachments
+        file_resources = {}
+        attachments = []
+
+        for resource in self.experiment.assistant.tool_resources.all():
+            for file in resource.files.all():
+                if file.external_id in file_resources:
+                    file_resources[file.external_id].append(resource)
+                else:
+                    file_resources[file.external_id] = [resource.tool_type]
+
+        for file_id, resources in file_resources.items():
+            tools = [{"type": resource} for resource in resources]
+            attachment = {"file_id": file_id, "tools": tools}
+            attachments.append(attachment)
+
+        input_dict["attachments"] = attachments
```

Comment:
This isn't necessary. The attachments here are per-message attachments which we don't support.

Reply:
Of course, we don't have to pass it along with each message. I wasn't thinking straight. Thanks!